var/spack/repos/builtin/packages/fsl/package.py

```diff
@@ -8,49 +8,178 @@
+                                  r'\1 = {0}'.format(
+                                      join_path(
+                                          cuda_installation,
+                                          'include')))
+            build_settings.filter(r'(^NVCC11)\s*=.*',
+                                  r'\1 = {0}'.format(
+                                      join_path(
+                                          cuda_installation, 'bin',
+                                          'nvcc')))
+            build_settings.filter(r'(^NVCC)\s*=.*',
+                                  r'\1 = {0}'.format(
+                                      join_path(
+                                          cuda_installation, 'bin',
+                                          'nvcc')))
+            build_settings.filter(r'(^GENCODE_FLAGS)\s*=.*',
+                                  r'\1 = {0}'.format(cuda_gencode))
+
+            if self.spec.satisfies('@6:'):
+                build_settings.filter(r'(^EDDYBUILDPARAMETERS)\s*=.*',
+                                      r'\1 = "cuda=1" "cpu=1"')
+                build_settings.filter(r'(^fdt_MASTERBUILD)\s*=.*',
+                                      r'\1 = COMPILE_GPU=1')
+                build_settings.filter(r'(^ptx2_MASTERBUILD)\s*=.*',
+                                      r'\1 = COMPILE_GPU=1')
+            else:
+                with open(settings_file, 'a') as f:
+                    f.write('COMPILE_GPU=1\n')
+        else:
+            build_settings.filter(r'^CUDA_INSTALLATION', '#CUDA_INSTALLATION')
+            build_settings.filter(r'^GENCODE_FLAGS', '#GENCODE_FLAGS')
+            build_settings.filter(r'^LIB_CUDA', '#LIB_CUDA')
+            build_settings.filter(r'^INC_CUDA', '#INC_CUDA')
+            build_settings.filter(r'^NVCC', '#NVCC')
+
+            if self.spec.satisfies('@6:'):
+                build_settings.filter(r'(^EDDYBUILDPARAMETERS)\s*=.*',
+                                      r'\1 = "cpu=1"')
+                build_settings.filter(r'(^fdt_MASTERBUILD)\s*=.*',
+                                      r'\1 = COMPILE_GPU=0')
+                build_settings.filter(r'(^ptx2_MASTERBUILD)\s*=.*',
+                                      r'\1 = COMPILE_GPU=0')
 
     def install(self, spec, prefix):
         build = Executable('./build')
         build()
 
         install_tree('.', prefix)
 
+    @run_after('install')
+    def postinstall(self):
+        # The PYTHON  related environment variables need to be unset here so
+        # the post install script does not get confused.
+        vars_to_unset = ['PYTHONPATH', 'PYTHONHOME']
+
+        with spack.util.environment.preserve_environment(*vars_to_unset):
+            for v in vars_to_unset:
+                del os.environ[v]
+
+        script = Executable(join_path(prefix, 'etc', 'fslconf',
+                                      'post_install.sh'))
+        script('-f', prefix)
```

Comment:
I think these two lines need to be indented one level more: the variables are deleted only within the context manager, but will be restored outside of it.

Reply:
Yes, sorry, I meant to do that but missed it as I was trying to understand the context manager. I will push out the fix.